packages/btfs-http-client/examples/upload-file-via-browser/src/App.js

```diff
@@ -1,92 +1,287 @@
-/* eslint-disable no-console */
 'use strict'
 const React = require('react')
-const ipfsClient = require('ipfs-http-client')
+const btfsClient = require('../../../../btfs-http-client/src/index')
+const config = require("../config")
+const session = require("../../../src/session/session")
+var errorCount = 0
 
 class App extends React.Component {
+  //BTFS-1437
   constructor () {
     super()
+
     this.state = {
-      added_file_hash: null
+      added_file_hash: null,
+      added_session_id:  null,
+      added_session_status: null,
+      added_session_contracts: null,
+      added_status_response: null,
+      added_session: new session(config.PrivKey, config.PeerID),
     }
-    this.ipfs = ipfsClient('/ip4/127.0.0.1/tcp/5001')
+
+    this.btfs = btfsClient('/ip4/127.0.0.1/tcp/5001')
 
     // bind methods
     this.captureFile = this.captureFile.bind(this)
     this.saveToIpfs = this.saveToIpfs.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
+    //bing offline signing methods
+    this.upload = this.upload.bind(this)
+    this.displayStatus = this.displayStatus.bind(this)
+    this.sign = this.sign.bind(this)
+    this.getBatch = this.getBatch.bind(this)
+    this.getUnsignedData = this.getUnsignedData.bind(this)
+
+    this.statusTimer = null
+    this.stateTimer = null
+    this.time = Date.now()
+
   }
 
-  captureFile (event) {
-    event.stopPropagation()
-    event.preventDefault()
-    if (document.getElementById('keepFilename').checked) {
-      this.saveToIpfsWithFilename(event.target.files)
-    } else {
-      this.saveToIpfs(event.target.files)
+  setTime(){
+    this.time = Date.now()
+  }
+
+  addStatus(status) {
+    let div = document.getElementById('contractStatus')
+    if (this.state.added_status_response != null){
+      div.innerHTML += "<h3>".concat(status).concat(": ").concat(this.state.added_status_response).concat("</h3>")
```

Comment:
use string template instead of `.concat()`, e.g. ``<h3>${status}: ${this.state.added_status_response}</h3>``

Reply:
ok.